### PR TITLE
fix mobile oauth tests

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
Fixes failing tests introduced in #24554 

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-54489

```release-note
NONE
```

### CODE REVIEW

**Comment by enahum** [comment]
Failures seems unrelated to this PR.

**Comment by harshilsharma63** [comment]
> Failures seems unrelated to this PR.

If I go exactly one commit before when you merged the PR into main, the tests work. And exactly on the PR merge commit they fail.

**Comment by enahum** [comment]
@harshilsharma63 are you able to see what the error is? I can't seem to find the details on the GHA

**Review by hmhealey — APPROVED** [review]

**Comment by hmhealey** [comment]
I'm not able to see the logs on the failed CI steps either. That's very bizarre. I posted in the Dev channel to see if anyone else has any other ideas

File: server/channels/web/oauth_test.go [hunk 1]
```diff
@@ -404,7 +404,7 @@ func TestMobileLoginWithOAuth(t *testing.T) {
 	var siteURL = "http://localhost:8065"
 	th.App.UpdateConfig(func(cfg *model.Config) {
 		*cfg.ServiceSettings.SiteURL = siteURL
-		cfg.NativeAppSettings.AppCustomURLSchemes = append(cfg.NativeAppSettings.AppCustomURLSchemes, "mmauth://")
+		*cfg.GitLabSettings.Enable = true
```
Comment (lieut-data) [inline server/channels/web/oauth_test.go:407]: The addition of `GitLabSettings.Enable` seems unexpected relative to https://github.com/mattermost/mattermost/pull/24554 -- is it intentional?

File: server/channels/web/oauth_test.go [hunk 2]
```diff
@@ -417,32 +417,36 @@ func TestMobileLoginWithOAuth(t *testing.T) {
 		responseWriter := httptest.NewRecorder()
 		request, _ := http.NewRequest(http.MethodGet, th.App.GetSiteURL()+"/oauth/gitlab/mobile_login?redirect_to="+url.QueryEscape("mmauth://"), nil)
 		mobileLoginWithOAuth(c, responseWriter, request)
-		assert.Contains(t, responseWriter.Body.String(), "mmauth://")
+		assert.Equal(t, responseWriter.Code, 302)
```
Comment (lieut-data) [inline server/channels/web/oauth_test.go:420]: The test case description suggests the redirect URL should still be present in the output -- should we update the description, or add back the `mmauth://` check?

**Review by lieut-data — CHANGES_REQUESTED** [review]
Jumping in with a review as I ran into this a pull request of mine.

File: server/channels/web/oauth_test.go (hunk 1 above)
Comment (enahum) [reply server/channels/web/oauth_test.go:407]: Yes without it, the method returns an error that no sso service is enabled

File: server/channels/web/oauth_test.go (hunk 2 above)
Comment (enahum) [reply server/channels/web/oauth_test.go:420]: Will update the description as the method will redirect to the sso login page when the scheme is valid

**Review by lieut-data — APPROVED** [review]
Sweet!

**Pull request merged**